### PR TITLE
Add google analytics JS

### PR DESCRIPTION
**Pull request by MatMoore**
This is conditional on an environment variable being set.

Once we have a cookie banner / cookie page, we can also skip rendering the javascript if analytics cookies are not accepted.

### CODE REVIEW

File: core/settings.py [hunk 1]
```diff
@@ -171,6 +174,8 @@
     },
 }
 
+ENABLE_ANALYTICS = os.environ.get("ENABLE_ANALYTICS") in TRUTHY_VALUES
```
Comment (tom-webber) [inline core/settings.py:177]:
```suggestion
ENABLE_ANALYTICS: bool = os.environ.get("ENABLE_ANALYTICS", "") in TRUTHY_VALUES
```

File: templates/base/base.html [hunk 2]
```diff
@@ -60,5 +60,9 @@
     </script>
     {% block scripts %}
     {% endblock scripts %}
+    {% if enable_analytics %}
+    <script async src="https://www.googletagmanager.com/gtag/js?id={{analytics_id}}"></script>
+    <script>window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag('js', new Date()); gtag('config', '{{analytics_id}}');</script>
+    {% endif %}
```
Comment (tom-webber) [inline templates/base/base.html:66]: ⛏️ minor suggestion for emphasising environment variable usage
```suggestion
    <script async src="https://www.googletagmanager.com/gtag/js?id={{ ANALYTICS_ID }}"></script>
    <script>window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag('js', new Date()); gtag('config', '{{ ANALYTICS_ID }}');</script>
    {% endif %}
```

File: core/settings.py [hunk 3]
```diff
@@ -171,6 +174,11 @@
     },
 }
 
+ANALYTICS_ID: str = os.environ.get("ANALYTICS_ID", "")
+ENABLE_ANALYTICS: bool = (
+    os.environ.get("ENABLE_ANALYTICS") in TRUTHY_VALUES
+) and ANALYTICS_ID != ""
```
[the remark this answers is not in the document]
Comment (MatMoore) [reply core/settings.py:180]: This isn't needed, `.get` defaults to None which is not in the list

File: templates/base/base.html (hunk 2 above)
Comment (MatMoore) [reply templates/base/base.html:66]: this needs to match the context processor. I've kept it lowercase to be consistent with how we've done the env variable

```
{"env": settings.ENV}
```

**Review by tom-webber — APPROVED** [review]
Nice!

**Pull request merged**